git-link.el

```diff
@@ -411,12 +428,30 @@ return (FILENAME . REVISION) otherwise nil."
                                 (format "L%s-L%s" start end)
                               (format "L%s" start)))))))
 
+(defun git-link-azure (hostname dirname filename branch commit start end)
+  (format "https://%s/%s?path=%s&version=%s%s"
+	  hostname
+	  dirname
+      filename
+      (concat "G" (if branch "B" "C") (or branch commit))
+      (concat
+       (when start (format "&line=%s" start))
+       (when end (format "&lineEnd=%s" end)))))
+
 (defun git-link-commit-github (hostname dirname commit)
   (format "https://%s/%s/commit/%s"
 	  hostname
 	  dirname
 	  commit))
 
+(defun git-link-commit-azure (hostname dirname commit)
+  (format "https://%s/%s/commit/%s"
+	  hostname
+	  dirname
+	  (with-temp-buffer
+        (vc-git--out-ok "rev-parse" commit)
+        (string-trim (buffer-string)))))
```

Comment:
Why is the call to `buffer-string` needed? The functions should depend on their args and not on the caller's context.